localstack/services/apigateway/helpers.py

```diff
@@ -896,6 +893,21 @@ def extract_api_id_from_hostname_in_url(hostname: str) -> str:
     return match.group(1)
 
 
+# This need to be extended to handle mappings and not just literal values.
+def create_invocation_headers(invocation_context: ApiInvocationContext) -> Dict[str, Any]:
+    headers = invocation_context.headers
+    integration = invocation_context.integration
+
+    if request_parameters := integration.get("requestParameters"):
+        for req_parameter_key, req_parameter_value in request_parameters.items():
+            if header_name := list(
```

Comment:
nit: Curious about the use of `list`/`filter` here - seems like we may need to adjust the logic a bit, if I'm not mistaken. If the value of `req_parameter_key` is, say, `"integration.request.path.foobar"`, then we'd update the `headers` with this key, which is probably not desired..

Reply:
Correct, the split doesn't work as I expected when the prefix doesn't match I wasn't expecting a return but in fact it returns the original value. Fixed now!